Unsupported type MethodAccessorGenerator reachable error

quarkus master (bbb2329ae) w/ graal master (https://github.com/oracle/graal/commit/ee232b09e21e) failing with Java 11 in `integration-tests/optaplanner-jackson`:

```
$ mvn clean install -Dnative -pl integration-tests/optaplanner-jackson
...
Error: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported type jdk.internal.reflect.MethodAccessorGenerator is reachable
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Detailed message:
Trace:
	at parsing jdk.internal.reflect.ReflectionFactory.generateConstructor(ReflectionFactory.java:513)
Call path from entry point to jdk.internal.reflect.ReflectionFactory.generateConstructor(Class, Constructor):
	at jdk.internal.reflect.ReflectionFactory.generateConstructor(ReflectionFactory.java:513)
	at jdk.internal.reflect.ReflectionFactory.newConstructorForSerialization(ReflectionFactory.java:506)
	at java.io.ObjectStreamClass.getSerializableConstructor(ObjectStreamClass.java:1516)
	at java.io.ObjectStreamClass$2.run(ObjectStreamClass.java:509)
	at java.io.ObjectStreamClass$2.run(ObjectStreamClass.java:484)
	at com.oracle.svm.core.jdk.Target_java_security_AccessController.doPrivileged(SecuritySubstitutions.java:81)
	at sun.security.pkcs11.SunPKCS11.initToken(SunPKCS11.java:1028)
	at sun.security.pkcs11.SunPKCS11$TokenPoller.run(SunPKCS11.java:852)
	at java.lang.Thread.run(Thread.java:834)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:517)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:193)
	at com.oracle.svm.core.code.IsolateEnterStub.PosixJavaThreads_pthreadStartRoutine_e1f4a8c0039f8337338252cd8734f63a79b5e3df(generated:0)
```

Console output can be found [here](https://gist.github.com/galderz/b57a0d7f8214348649c2541fa488c8f2).

I've tried to do a basic substitution:

```java
@TargetClass(java.io.ObjectStreamClass.class)
final class Target_java_io_ObjectStreamClass {

    @Alias private Long suid;

    @Substitute
    public long getSerialVersionUID() {
        if (suid != null) {
            return suid;
        }

        return 1L;
    }
}
```

But that's not enough, it would then fail with:

```

Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: com.oracle.svm.hosted.substitute.DeletedElementException: Unsupported type jdk.internal.reflect.MethodAccessorGenerator is reachable
To diagnose the issue, you can add the option --report-unsupported-elements-at-runtime. The unsupported element is then reported at run time when it is accessed the first time.
Detailed message:
Trace:
	at parsing jdk.internal.reflect.ReflectionFactory.generateConstructor(ReflectionFactory.java:513)
Call path from entry point to jdk.internal.reflect.ReflectionFactory.generateConstructor(Class, Constructor):
	at jdk.internal.reflect.ReflectionFactory.generateConstructor(ReflectionFactory.java:513)
	at jdk.internal.reflect.ReflectionFactory.newConstructorForSerialization(ReflectionFactory.java:506)
	at java.io.ObjectStreamClass.getSerializableConstructor(ObjectStreamClass.java:1516)
	at java.io.ObjectStreamClass$2.run(ObjectStreamClass.java:509)
	at java.io.ObjectStreamClass$2.run(ObjectStreamClass.java:484)
	at com.oracle.svm.core.jdk.Target_java_security_AccessController.doPrivileged(SecuritySubstitutions.java:81)
	at jdk.internal.module.IllegalAccessLogger.codeSource(IllegalAccessLogger.java:325)
	at jdk.internal.module.IllegalAccessLogger.lambda$log$1(IllegalAccessLogger.java:262)
	at jdk.internal.module.IllegalAccessLogger$$Lambda$898/0x00000007c1b68840.get(Unknown Source)
	at com.oracle.svm.core.jdk.SystemPropertiesSupport.initializeLazyValue(SystemPropertiesSupport.java:190)
	at com.oracle.svm.core.jdk.SystemPropertiesSupport.getProperty(SystemPropertiesSupport.java:143)
	at com.oracle.svm.core.jdk.Target_java_lang_System.getProperty(JavaLangSubstitutions.java:340)
	at com.oracle.svm.jni.JNIJavaCallWrappers.jniInvoke_ARRAY:Ljava_lang_System_2_0002egetProperty_00028Ljava_lang_String_2_00029Ljava_lang_String_2(generated:0)
```

--- a/extensions/optaplanner/runtime/src/main/java/io/quarkus/optaplanner/runtime/graal/XStreamSubstitutions.java
+++ b/extensions/optaplanner/runtime/src/main/java/io/quarkus/optaplanner/runtime/graal/XStreamSubstitutions.java
@@ -1,0 +1,52 @@
+package io.quarkus.optaplanner.runtime.graal;
+
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+@TargetClass(className = "com.thoughtworks.xstream.converters.reflection.SerializableConverter")
+final class Target_SerializableConverter {
+
+    @Substitute
+    public Object doUnmarshal(final Object result, final Target_HierarchicalStreamReader reader,
+            final Target_UnmarshallingContext context) {
+        return null;
+    }
+
+    @Substitute
+    public void doMarshal(final Object source, final Target_HierarchicalStreamWriter writer,
+            final Target_MarshallingContext context) {
+    }
+}
+
+@TargetClass(className = "com.thoughtworks.xstream.io.HierarchicalStreamReader")
+final class Target_HierarchicalStreamReader {
+
+}
+
+@TargetClass(className = "com.thoughtworks.xstream.converters.UnmarshallingContext")
+final class Target_UnmarshallingContext {
+
+}
+
+@TargetClass(className = "com.thoughtworks.xstream.io.HierarchicalStreamWriter")
+final class Target_HierarchicalStreamWriter {
+
+}
+
+@TargetClass(className = "com.thoughtworks.xstream.converters.MarshallingContext")
+final class Target_MarshallingContext {
+
+}
+
+@TargetClass(className = "com.thoughtworks.xstream.converters.reflection.PureJavaReflectionProvider")
+final class Target_PureJavaReflectionProvider {
+
+    @Substitute
+    private Object instantiateUsingSerialization(final Class type) {
+        return null;
+    }
+}
+
+class XStreamSubstitutions {
+
+}

